src/metabase/models/session.clj

```diff
@@ -71,21 +82,34 @@
   [:and
    [:map-of :keyword :any]
    [:map
-    [:id uuid?]
+    [:key string?]
     [:type [:enum :normal :full-app-embed]]]])
 
 (defmulti create-session!
   "Generate a new Session for a User. `session-type` is currently either `:password` (for email + password login) or
-  `:sso` (for other login types). Returns the newly generated Session."
+  `:sso` (for other login types). Returns the newly generated Session with the id as the plain-text session-key."
   {:arglists '([session-type user device-info])}
   (fn [session-type & _]
     session-type))
 
+(defn generate-session-key
+  "Generate a new session key."
+  []
+  (str (random-uuid)))
+
+(defn generate-session-id
+  "Generate a new id for the session table."
+  []
+  (string/random-string 12))
```

Comment:
The definition of `random-string` threw me for a second because it uses the non-cryptographically-secure `rand-nth` instead of something like `SecureRandom`. Might be worth adding a comment here that this *doesn't* need to be cryptographically secure (because it'll no longer be a token), and a comment in `random-string` that it shouldn't be used for crypto?